hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/schema/DataConnectionResolver.java

```diff
@@ -147,10 +152,18 @@ public static List<List<?>> getAllDataConnectionNameWithTypes(
         conn.addAll(dataConnectionService.getSqlCreatedDataConnections());
 
         return conn.stream()
-                   .map(dc -> Arrays.asList(dc.getName(), new ArrayList<>(dc.resourceTypes())))
+                   .map(dc -> asList(dc.getName(), json(dc.resourceTypes())))
                    .collect(toList());
     }
 
+    private static HazelcastJsonValue json(Object value) {
+        try {
+            return new HazelcastJsonValue(SERIALIZER.writeValueAsString(value));
```

Comment:
I have doubts regarding that part. Did you try `JsonArray` ?

Reply:
We have three different JSON serializers: `JsonCreationUtil`, which uses Gson internally; Jackson `ObjectMapper`, which is used in tests; and finally `JsonValue.toString()`. I switched to the last one to get rid of external dependencies, but I don't know its performance characteristics.